Eval frame prune rebase

bcolloran opened the pull request:
### Pull Request checklist
<!-- Before submitting the PR, please address each item -->
- [x] **Documentation**: behind the scenes -- not needed
- [x] **Changelog**: behind the scenes -- not needed
- [ ] **Tests**: there are some test regressions here, and no new tests, but a lot of this code is scaffolding code that will be replaced as i move stuff around, so i'm not ready to commit to tests yet. yes i know that's considerd backwards by many, but...

bcolloran commented on src/eval-frame/actions/actions.js:
```diff
@@ -1,17 +1,10 @@
 
 import { IODIDE_EVALUATION_RESULTS } from "../iodide-evaluation-results";
 
-const CodeMirror = require("codemirror"); // eslint-disable-line
```
this is all just dead code that has not been needed in the eval frame for a long time

bcolloran commented on src/eval-frame/components/reps/__stories__/reps.stories.js:
```diff
@@ -214,44 +206,3 @@ allTestCases.add("expandable rep", () => {
     </div>
   );
 });
-
```
changes in this file are just about moving some stories to their own files and cleaning things up

bcolloran commented on src/eval-frame/components/reps/__test_helpers__/reps-test-value-cases.js:
```diff
@@ -318,24 +318,39 @@ export const promiseCases = {
 
 export const rowTableCases = {
-  rowsTable_plainObjects: new Array(100).fill(0).map((x, i) => ({
+  rowsTable_plainObjects: new Array(58).fill(0).map((x, i) => ({
```
changed a bunch of the text examples so they aren't all just 100 -- we should test more than just nice round numbers

bcolloran commented on src/eval-frame/components/reps/__test_helpers__/reps-test-value-cases.js:
```diff
@@ -318,24 +318,39 @@ export const promiseCases = {
   })),
-  rowsTable_compositeObjects: new Array(100).fill(compositeObjects),
-  rowsTable_objectsContainingSimpleTypes: new Array(100).fill(simpleTypes),
+
```
added a couple cases of tables that don't just repeat the same rows over and over

bcolloran commented on src/eval-frame/components/reps/__test_helpers__/reps-test-value-cases.js:
```diff
@@ -347,6 +362,24 @@ export const blobObjects = {
     type: "application/json"
   })
 };
+
```
test case for objects with `iodideRender()` method

bcolloran commented on src/eval-frame/components/reps/__tests__/value-renderer.test.js:
```diff
@@ -7,25 +7,14 @@ import ValueRenderer from "../value-renderer";
+import TableRenderer from "../data-table-rep";
 
-import { IODIDE_EVALUATION_RESULTS } from "../../../iodide-evaluation-results";
+// FIXME: refactor ValueRenderer to make testing more tractable.
```
i had to basically gut this test for now. not ideal, but for the moment i think the reduced automatic coverage is ok b/c this is all covered by the extensive storybook

bcolloran commented on src/eval-frame/components/reps/data-table-rep.jsx:
```diff
@@ -1,17 +1,14 @@
 
-import { getChildSummary } from "./rep-utils/get-child-summaries";
-import { RangeDescriptor } from "./rep-utils/rep-serialization-core-types";
+import { requestRepInfo } from "./request-rep-info";
```
all the changes in this file are about using `requestRepInfo` rather than direct functions for serializing object descriptors. this is the key stepping stone that will eventually allow truly remote reps across iframes.

bcolloran commented on src/eval-frame/components/reps/error-handler.jsx:
```diff
@@ -2,129 +2,6 @@ import React from "react";
 import styled from "react-emotion";
 import PropTypes from "prop-types";
 
-import ErrorStackParser from "error-stack-parser";
```
broke the error stack parser into a separate file from the component so that it can be used remotely

bcolloran commented on src/eval-frame/components/reps/rep-utils/get-data-table-summary.js:
```diff
@@ -0,0 +1,44 @@
+
+import { getObjAtPath } from "./get-child-summaries";
+
+export function getDataTableSummary(rootObjName, path, pageSize, page) {
```
this file/function is about producing the summary data to show a tabular rep remotely

bcolloran commented on src/eval-frame/components/reps/rep-utils/get-error-stack-summary.js:
```diff
@@ -0,0 +1,122 @@
+import ErrorStackParser from "error-stack-parser";
+import StackFrame from "stackframe";
+
+ErrorStackParser.parseV8OrIE = error => {
```
this file is just code moved over from the error rep component

bcolloran commented on src/eval-frame/components/reps/rep-utils/get-top-level-rep-summary.js:
```diff
@@ -0,0 +1,33 @@
+
+import UserReps from "../user-reps-manager";
+
+export function getTopLevelRepSummary(rootObjName, path) {
```
this file/function produces the top-level summary needed to chose between different top level reps

bcolloran commented on src/eval-frame/components/reps/value-renderer.jsx:
```diff
@@ -6,56 +6,61 @@ import ExpandableRep from "./rep-tree";
-  }
-  return "DEFAULT_REP";
-}
+import { requestRepInfo } from "./request-rep-info";
```
all the changes in this file are about using `requestRepInfo` rather than direct functions for serializing object descriptors. this is the key stepping stone that will eventually allow truly remote reps across iframes.

bcolloran commented on src/eval-frame/components/reps/rep-utils/rep-serialization-core-types.js:
```diff
@@ -19,7 +19,8 @@ const nonExpandableTypes = [
   "ArrayBuffer",
-  "DataView"
+  "DataView",
+  "Blob"
```
blobs should not expand. this just fixes a small bug i noticed.

bcolloran commented on src/eval-frame/components/reps/rep-utils/rep-serialization-core-types.js:
```diff
@@ -62,6 +63,7 @@ export class ValueSummary {
     this.size = size; // number
     this.stringValue = stringValue; // string
     this.isTruncated = isTruncated; // bool
+    this.DESC_TYPE = "ValueSummary";
```
ignore all the changes like this in this file. just laying groundwork for later.

bcolloran commented on src/eval-frame/components/reps/request-rep-info.js:
```diff
@@ -0,0 +1,5 @@
+import { repInfoRequestResponse } from "./rep-utils/rep-info-request-response";
+
+export async function requestRepInfo(payload) {
```
this is a stub for proper cross frame request machinery

bcolloran commented:
@hamilton: i'd like to request speedy review on this -- i'm making reasonable progress on the eval-frame disentangling and i'd like to keep the momentum up. since i'll have to touch a lot of files in this process, i want to merge small bits back to master rather than one giant PR that turns the whole world upside down.

i've left a bunch of notes to help expedite this review, but basically this is about doing more of the work needed to isolate the rep components from the object descriptor serializers. hopefully it's not too bad of a review. please try to get to this early thurs and LMK if anything looks nutso.

you can copy/paste the code in https://alpha.iodide.io/notebooks/2756/ to your local set up for QA if you wish.

hamilton commented:
@bcolloran sounds good to me – will review this today. Thanks for moving quickly on this.

hamilton commented on src/eval-frame/components/reps/data-table-rep.jsx:
```diff
@@ -105,57 +113,115 @@ const MIN_CELL_CHAR_WIDTH = 22;
-        PropTypes.string,
-        PropTypes.instanceOf(RangeDescriptor)
-      ])
+    initialDataRows: PropTypes.arrayOf(PropTypes.object).isRequired,
```
fyi – the column sort buttons in this PR don't work on the table rep

hamilton replied:
(disregard – you mention this in a FIXME elsewhere)

hamilton commented on src/eval-frame/components/reps/rep-utils/get-top-level-rep-summary.js:
```diff
@@ -0,0 +1,33 @@
+  if (UserReps.getUserRepIfAvailable(value)) {
+    return {
+      repType: "HTML_STRING",
+      htmlString: UserReps.getUserRepIfAvailable(value)
```
Just an FYI – the images here don't render

<img width="593" alt="Screen Shot 2019-07-18 at 4 47 26 PM" src="https://user-images.githubusercontent.com/95735/61499377-d0c2a700-a97b-11e9-8142-ddaddb960b7d.png">

bcolloran replied:
this is a problem with the API mike used for the example -- if you look closely you'll see a little magnifying glass icon, so the images that are being returned are rendering, they're just not great images.

that said, i'm not sure whether we'll want to support image tags or not in the long run -- def a tracking pixel vector, perhaps other security risks?

hamilton commented on src/eval-frame/components/reps/rep-utils/get-data-table-summary.js:
```diff
@@ -0,0 +1,44 @@
+export function getDataTableSummary(rootObjName, path, pageSize, page) {
+  const tabularObj = getObjAtPath(window[rootObjName], path);
+
+  // // FIXME: we may wish to consider making the tables sortable
```
oh whoops I missed this note when I made my other comment. Seems like this would be useful to make sure we have down the line.

bcolloran commented:
@hamilton i went ahead and added sorting back in and added storybook cases to test the behavior. i also moved some of the stuff around requesting the rep info to HOC to simplify testing, and I added some tests back in. going to merge now
